services/director/src/simcore_service_director/registry_proxy.py

```diff
@@ -102,13 +163,35 @@ class ServiceType(enum.Enum):
             repos_list.extend(result["repositories"])
         request = URL(str(headers["Link"]).split(";")[0].strip("<>")) if "Link" in headers else None
         loop = "Link" in headers
+    _logger.debug("listed %s repositories", len(repos_list))
     return repos_list
 
-async def _get_image_details(image_key: str, image_tag: str) -> Dict:
+async def list_image_tags(image_key: str) -> List[Dict]:
+    _logger.debug("listing image tags in %s", image_key)
+    image_tags = []
+    # get list of image tags
+    loop = True
+    request = URL("v2/{}/tags/list?n={}".format(image_key, NUMBER_OF_RETRIEVED_TAGS))
+    while loop:
```

Comment:
IMHO this reads better that using the variable ``loop`` which in this context typically refers to the event loop
```python
header = ""
while "Link" in header:
  header = ...
```

Reply:
goog catch! I actually need a do...while type of loop. so changed it to while True, if condition false break instead.